src/ucp/core/ucp_request.inl

```diff
@@ -54,12 +55,18 @@ ucp_request_complete_recv(ucp_request_t *req, ucs_status_t status,
     ucs_trace_data("completing recv request %p (%p) stag 0x%"PRIx64" len %zu, %s",
                    req, req + 1, info->sender_tag, info->length,
                    ucs_status_string(status));
-    req->recv.cb(req + 1, status, info);
+
+    if (req->flags & UCP_REQUEST_FLAG_EXTERNAL) {
+        req->flags |= UCP_REQUEST_FLAG_COMPLETED;
+        req->status = status;
+    } else {
+        req->recv.cb(req + 1, status, info);
```

Comment:
seems like doing ucp_request_test on an internal request would not return the correct status?